lab4/lvl3/lab4.py

```diff
@@ -0,0 +1,38 @@
+def read_from_file(file_name: str):
+    with open(file_name, "r") as file:
+        n = int(file.readline().strip())
+        print(n)
+        start = tuple(map(int, file.readline().strip().split(',')))
+        print(start)
+        destination = tuple(map(int, file.readline().strip().split(',')))
+        print(destination)
+    return n, start, destination
+
+
+def find_min_knight_path(board_side: int, start: tuple, destination: tuple):
+    row = [2, 2, -2, -2, 1, 1, -1, -1]
+    col = [-1, 1, 1, -1, 2, -2, 2, -2]
+    counter = 0
+    queue = [(start, counter)]
+    visited = set()
+    while queue:
+        (x, y), counter = queue.pop(0)
```

Comment:
to reduce packing `x` and `y` into tuple every time you want to use them together, you may write something like

```
current_position, counter = queue.pop(0)
x, y = current_position
```

and then use `current_position` when you need it, and `x` and `y` when you need them